.github/workflows/build-test-linux.yml

```diff
@@ -61,7 +48,10 @@ jobs:
     - name: Cache build output
       uses: actions/cache@v2
       with:
-        path: ${{ matrix.cachepath }}
+        path: |
+              ~/.cache/go-build
+              ~/go/pkg/mod
+              /home/runner/.cache/go-build
```

Comment:
We can delete the home runner cache based on the tests I have run: 
* Test with a PR and found unexpected results without caching this path runner/Library/Caches/go-build:
https://github.com/aws/amazon-cloudwatch-agent/runs/4990690788?check_suite_focus=true
https://github.com/aws/amazon-cloudwatch-agent/runs/4990745483?check_suite_focus=true

* Compared when cached this path runner/Library/Caches/go-build:
https://github.com/aws/amazon-cloudwatch-agent/runs/4990443776?check_suite_focus=true

Reply:
This can be the same when cache build packages for window and mac